### ISSUE TITLE
Does not work on CUDA

### ISSUE THREAD

**Issue by HEmile** [lucadiliello/semantic-loss-pytorch#2]
This code does not run on CUDA. Here's a simple fix (I can pull request if needed):

In sdd.py, use:
```
        true_tensor = torch.ones(1, device=trueprobs.device)
        false_tensor = torch.zeros(1, device=trueprobs.device)
```
On lines 549 and 550. There's also another fix needed for multi-gpu support.

**Comment by lucadiliello** [comment]
If you could open a PR it would be great!
